nexus/db-queries/src/db/datastore/rack.rs

```diff
@@ -109,6 +187,247 @@ impl DataStore {
+                    Uuid::new_v4(),
+                    &db::model::Name(nic.name.clone()),
+                    &format!("{}", service.kind),
+                    service.service_id,
+                    service_pool.id(),
+                    external_address,
+                );
+                let vpc_subnet = match service.kind {
+                    ServiceKind::ExternalDns { .. } => DNS_VPC_SUBNET.clone(),
+                    ServiceKind::Nexus { .. } => NEXUS_VPC_SUBNET.clone(),
+                    _ => unreachable!(),
+                };
+                let db_nic = IncompleteNetworkInterface::new_service(
+                    nic.id,
+                    service.service_id,
+                    vpc_subnet,
+                    IdentityMetadataCreateParams {
+                        name: nic.name.clone(),
+                        description: format!("{} service vNIC", service.kind),
+                    },
+                    Some(nic.ip),
+                    Some(nic.mac),
+                )
+                .map_err(|e| {
+                    TxnError::CustomError(RackInitError::AddingNic(e))
+                })?;
+                Some((external_address, db_ip, db_nic))
+            }
+            ServiceKind::BoundaryNtp { snat, ref nic } => {
+                let db_ip = IncompleteExternalIp::for_service_explicit_snat(
+                    Uuid::new_v4(),
+                    service.service_id,
+                    service_pool.id(),
+                    snat.ip,
+                    (snat.first_port, snat.last_port),
+                );
+                let db_nic = IncompleteNetworkInterface::new_service(
+                    nic.id,
+                    service.service_id,
+                    NTP_VPC_SUBNET.clone(),
+                    IdentityMetadataCreateParams {
+                        name: nic.name.clone(),
+                        description: format!("{} service vNIC", service.kind),
+                    },
+                    Some(nic.ip),
+                    Some(nic.mac),
+                )
+                .map_err(|e| {
+                    TxnError::CustomError(RackInitError::AddingNic(e))
+                })?;
+                Some((snat.ip, db_ip, db_nic))
+            }
+            _ => None,
+        };
+        if let Some((external_ip, db_ip, db_nic)) = service_ip_nic {
+            let allocated_ip =
+                Self::allocate_external_ip_on_connection(conn, db_ip)
+                    .await
+                    .map_err(|err| {
+                        warn!(
+                            log,
+                            "Initializing Rack: Failed to allocate \
+                            IP address for {}",
+                            service.kind,
+                        );
+                        TxnError::CustomError(RackInitError::AddingIp(err))
+                    })?;
+            assert_eq!(allocated_ip.ip.ip(), external_ip);
```

Comment:
This was already there but I'm inclined to just remove it instead of error'ing. We already have tests for ensuring we get back the requesting IP or error'ing.